test/new-e2e/examples/vmenv_withos_test.go

```diff
@@ -8,21 +8,51 @@ package examples
 import (
 	"testing"
 
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
-	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2os"
-	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/test"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/DataDog/test-infra-definitions/components/vm"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2vm"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 	"github.com/stretchr/testify/assert"
 )
 
-type vmSuiteEx1 struct {
-	e2e.Suite[e2e.VMEnv]
+type Remote vm.ERemote
+
+var _ components.Importable = &Remote{}
+
+func (r *Remote) GetClient(t *testing.T) (*client.VMClient, error) {
+	return client.NewVMClient(t, &utils.Connection{
+		Host: r.Host,
+		User: r.Username,
+	}, r.OSType)
+}
+
+type myVMEnv struct {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Something you remove with this implementation is the boundary around an environment. It allows exporting anything, but it does not prevent errors where we forget to export something. 

We miss on the protection layer granted by reflection in [`CheckEnvStructValid`](https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/pkg/utils/e2e/client/pulumi_stack_initializer.go#L25-L29), but maybe I missed it and it was moved somewhere